SystemInformer/delayhook.c

```diff
@@ -1230,62 +1457,122 @@ HRESULT PhDrawThemeBackgroundHook(
+)
+{
+    WCHAR className[MAX_PATH];
+
+    // Apply theme to ListView checkboxes
+    if (iPartId == BP_CHECKBOX /*|| iPartId == BP_RADIOBUTTON*/)
+    {
+        if (PhGetThemeClass(hTheme, className, RTL_NUMBER_OF(className)) &&
+            PhEqualStringZ(className, VSCLASS_BUTTON, TRUE))
+        {
+            HTHEME darkButtonTheme = PhOpenThemeData(NULL, L"DarkMode_Explorer::Button", 0);
+            HRESULT retVal = DefaultDrawThemeBackgroundEx(darkButtonTheme ? darkButtonTheme : hTheme, hdc, iPartId, iStateId, pRect, pOptions);
+            if (darkButtonTheme)
+                PhCloseThemeData(darkButtonTheme);
+            return retVal;
+        }
+    }
 
-        if (PhGetThemeClass(Theme, className, RTL_NUMBER_OF(className)))
+    // Micro optimization
+    if ((iPartId == TDLG_PRIMARYPANEL || iPartId == TDLG_FOOTNOTEPANE || iPartId == TDLG_SECONDARYPANEL || iPartId == TDLG_FOOTNOTESEPARATOR || iPartId == TDLG_EXPANDOBUTTON) &&
+        PhGetThemeClass(hTheme, className, RTL_NUMBER_OF(className)) && PhEqualStringZ(className, VSCLASS_TASKDIALOG, TRUE)
+        /*|| WindowsVersion < WINDOWS_11 && WindowFromDC(hdc) == NULL*/)
+    {
+        switch (iPartId)
         {
-            if (PhEqualStringZ(className, VSCLASS_MENU, TRUE))
+        case TDLG_PRIMARYPANEL:
+            SetDCBrushColor(hdc, PhThemeWindowBackground2Color);
+            FillRect(hdc, pRect, PhGetStockBrush(DC_BRUSH));
+            return S_OK;
+        case TDLG_FOOTNOTEPANE:
+            FillRect(hdc, pRect, PhThemeWindowBackgroundBrush);
+            return S_OK;
+        case TDLG_SECONDARYPANEL:
+        {
+            FillRect(hdc, pRect, PhThemeWindowBackgroundBrush);
+            RECT rect = *pRect;
+            rect.bottom = rect.top + 1;
+            SetDCBrushColor(hdc, PhThemeWindowForegroundColor);
+            FillRect(hdc, &rect, PhGetStockBrush(DC_BRUSH));
+            PhOffsetRect(&rect, 0, 1);
+            SetDCBrushColor(hdc, PhThemeWindowBackground2Color);
+            FillRect(hdc, &rect, PhGetStockBrush(DC_BRUSH));
+            return S_OK;
+        }
+        case TDLG_FOOTNOTESEPARATOR:
+        {
+            SetDCBrushColor(hdc, PhThemeWindowForegroundColor);
+            FillRect(hdc, pRect, PhGetStockBrush(DC_BRUSH));
+            RECT rect = *pRect;
+            rect.top += 1;
+            SetDCBrushColor(hdc, PhThemeWindowBackground2Color);
+            FillRect(hdc, &rect, PhGetStockBrush(DC_BRUSH));
+            return S_OK;
+        }
+        case TDLG_EXPANDOBUTTON:
             {
-                if (PartId == MENU_POPUPGUTTER || PartId == MENU_POPUPBORDERS)
+                // In Windows 11, buttons lack background, making them indistinguishable on dark backgrounds.
+                // To address this, we invert the button. This technique isn't applicable to Windows 10 as it causes the button's border to appear chipped.
+                static enum { yes, no, unknown } mustInvertButton = unknown;
```

Comment:
I see a comment indicating intention to restrict this to Windows 10, but not runtime check?